### ISSUE TITLE
app crash when trying to view or download any text file

### ISSUE THREAD

**Issue by szanzibar** [haiwen/seadroid#592]
If I either tap on on a text file or try to download it, the app crashes. I've tried deleting all appdata, uninstalling, different text files but still am getting the same thing. I tried creating a text file on my phone, uploading it, and it looks like if it is already cached I am able to open it just fine.

I get this in the log:

```
10-28 20:00:34.727 E/ConcurrentAsyncTask(16836): Uncaught exception in thread pool
10-28 20:00:34.727 E/ConcurrentAsyncTask(16836): java.lang.RuntimeException: An error occurred while executing doInBackground()
10-28 20:00:34.727 E/ConcurrentAsyncTask(16836):    at android.os.AsyncTask$3.done(AsyncTask.java:309)
10-28 20:00:34.727 E/ConcurrentAsyncTask(16836):    at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:354)
10-28 20:00:34.727 E/ConcurrentAsyncTask(16836):    at java.util.concurrent.FutureTask.setException(FutureTask.java:223)
10-28 20:00:34.727 E/ConcurrentAsyncTask(16836):    at java.util.concurrent.FutureTask.run(FutureTask.java:242)
10-28 20:00:34.727 E/ConcurrentAsyncTask(16836):    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113)
10-28 20:00:34.727 E/ConcurrentAsyncTask(16836):    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588)
10-28 20:00:34.727 E/ConcurrentAsyncTask(16836):    at java.lang.Thread.run(Thread.java:818)
10-28 20:00:34.727 E/ConcurrentAsyncTask(16836): Caused by: java.lang.NumberFormatException: Invalid long: "null"
10-28 20:00:34.727 E/ConcurrentAsyncTask(16836):    at java.lang.Long.invalidLong(Long.java:124)
10-28 20:00:34.727 E/ConcurrentAsyncTask(16836):    at java.lang.Long.parseLong(Long.java:345)
10-28 20:00:34.727 E/ConcurrentAsyncTask(16836):    at java.lang.Long.parseLong(Long.java:321)
10-28 20:00:34.727 E/ConcurrentAsyncTask(16836):    at com.seafile.seadroid2.SeafConnection.getFileFromLink(SeafConnection.java:566)
10-28 20:00:34.727 E/ConcurrentAsyncTask(16836):    at com.seafile.seadroid2.SeafConnection.getFile(SeafConnection.java:678)
10-28 20:00:34.727 E/ConcurrentAsyncTask(16836):    at com.seafile.seadroid2.data.DataManager.getFile(DataManager.java:403)
10-28 20:00:34.727 E/ConcurrentAsyncTask(16836):    at com.seafile.seadroid2.transfer.DownloadTask.doInBackground(DownloadTask.java:71)
10-28 20:00:34.727 E/ConcurrentAsyncTask(16836):    at com.seafile.seadroid2.transfer.DownloadTask.doInBackground(DownloadTask.java:19)
10-28 20:00:34.727 E/ConcurrentAsyncTask(16836):    at android.os.AsyncTask$2.call(AsyncTask.java:295)
10-28 20:00:34.727 E/ConcurrentAsyncTask(16836):    at java.util.concurrent.FutureTask.run(FutureTask.java:237)
10-28 20:00:34.727 E/ConcurrentAsyncTask(16836):    ... 3 more
10-28 20:00:34.727 E/AndroidRuntime(16836): FATAL EXCEPTION: SeadroidAsyncTask #4
10-28 20:00:34.727 E/AndroidRuntime(16836): Process: com.seafile.seadroid2, PID: 16836
10-28 20:00:34.727 E/AndroidRuntime(16836): java.lang.RuntimeException: An error occurred while executing doInBackground()
10-28 20:00:34.727 E/AndroidRuntime(16836):     at android.os.AsyncTask$3.done(AsyncTask.java:309)
10-28 20:00:34.727 E/AndroidRuntime(16836):     at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:354)
10-28 20:00:34.727 E/AndroidRuntime(16836):     at java.util.concurrent.FutureTask.setException(FutureTask.java:223)
10-28 20:00:34.727 E/AndroidRuntime(16836):     at java.util.concurrent.FutureTask.run(FutureTask.java:242)
10-28 20:00:34.727 E/AndroidRuntime(16836):     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113)
10-28 20:00:34.727 E/AndroidRuntime(16836):     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588)
10-28 20:00:34.727 E/AndroidRuntime(16836):     at java.lang.Thread.run(Thread.java:818)
10-28 20:00:34.727 E/AndroidRuntime(16836): Caused by: java.lang.NumberFormatException: Invalid long: "null"
10-28 20:00:34.727 E/AndroidRuntime(16836):     at java.lang.Long.invalidLong(Long.java:124)
10-28 20:00:34.727 E/AndroidRuntime(16836):     at java.lang.Long.parseLong(Long.java:345)
10-28 20:00:34.727 E/AndroidRuntime(16836):     at java.lang.Long.parseLong(Long.java:321)
10-28 20:00:34.727 E/AndroidRuntime(16836):     at com.seafile.seadroid2.SeafConnection.getFileFromLink(SeafConnection.java:566)
10-28 20:00:34.727 E/AndroidRuntime(16836):     at com.seafile.seadroid2.SeafConnection.getFile(SeafConnection.java:678)
10-28 20:00:34.727 E/AndroidRuntime(16836):     at com.seafile.seadroid2.data.DataManager.getFile(DataManager.java:403)
10-28 20:00:34.727 E/AndroidRuntime(16836):     at com.seafile.seadroid2.transfer.DownloadTask.doInBackground(DownloadTask.java:71)
10-28 20:00:34.727 E/AndroidRuntime(16836):     at com.seafile.seadroid2.transfer.DownloadTask.doInBackground(DownloadTask.java:19)
10-28 20:00:34.727 E/AndroidRuntime(16836):     at android.os.AsyncTask$2.call(AsyncTask.java:295)
10-28 20:00:34.727 E/AndroidRuntime(16836):     at java.util.concurrent.FutureTask.run(FutureTask.java:237)
10-28 20:00:34.727 E/AndroidRuntime(16836):     ... 3 more
10-28 20:00:34.728 E/AbstractTracker(16836): Can't create handler inside thread that has not called Looper.prepare()
10-28 20:00:34.728 D/AppTracker(16836): App Event: crash
10-28 20:00:34.738 E/AbstractTracker(16836): mTrackerAsyncQueryHandler is null
10-28 20:00:34.740 W/ActivityManager(3377):   Force finishing activity com.seafile.seadroid2/.ui.activity.FileActivity
10-28 20:00:34.743 W/ActivityManager(3377):   Force finishing activity com.seafile.seadroid2/.ui.activity.BrowserActivity
10-28 20:00:34.750 D/AbstractTracker(16836): Event success
```

**Comment by Logan676** [comment]
This bug was due to https://github.com/haiwen/seadroid/blob/master/app/src/main/java/com/seafile/seadroid2/SeafConnection.java#L566, which was introuduced by 72b6d639e9919ec3c845d68ea704e79d51779d76.
Should add Null pointer check.

**Comment by tschiew** [comment]
Got the same problem :(
Do you know already when the fix will be available?

**Comment by szanzibar** [comment]
No idea. In the meantime I have downgraded to the previous version of the Android app which doesn't have the bug.

**Comment by lins05** [comment]
This bug is fixed in the just-release seadroid 2.1.7. Thanks for all reporting on this.

For those interested, I've briefly described the underlying reason of this bug in [this commit](https://github.com/haiwen/seadroid/commit/afae2da5a6c3b7cb4c01b62cc80ca03dbda11949).
